src/ecHiGoal.ml

```diff
@@ -163,17 +163,24 @@ let process_coq ~loc ~name (ttenv : ttenv) coqmode  pi (tc : tcenv1) =
     t_seq (t_simplify ~delta:`No) (t_coq ~loc ~name ~mode:(`Report (Some loc)) coqmode pi) tc
 
 (* -------------------------------------------------------------------- *)
-let process_clear symbols tc =
+let process_clear (info : clear_info) tc =
   let hyps = FApi.tc1_hyps tc in
-
   let toid s =
     if not (LDecl.has_name (unloc s) hyps) then
       tc_lookup_error !!tc ~loc:s.pl_loc `Local ([], unloc s);
     fst (LDecl.by_name (unloc s) hyps)
   in
-
-  try  t_clears (List.map toid symbols) tc
-  with (ClearError _) as err -> tc_error_exn !!tc err
+  match info with
+  | `Include symbols -> begin
+    try  t_clears (List.map toid symbols) tc
+    with (ClearError _) as err -> tc_error_exn !!tc err
+  end
+  | `Exclude symbols -> 
+    let try_clear tc id = try t_clear1 id tc with (ClearError _) -> tc in
```

Comment:
I think that the `leniant` option of `t_clear1` servers this purpose: not failing when the hypothesis cannot be cleared.

Reply:
You're right. I've switched to using `t_clears` with `leniant` set.